Module-4/Task3/UserTaskController.cs

```diff
@@ -1,42 +1,29 @@
-﻿using Task3.DoNotChange;
+﻿using System;
+using Task3.DoNotChange;
 
-namespace Task3
+namespace Task3;
+
+public class UserTaskController
 {
-    public class UserTaskController
-    {
-        private readonly UserTaskService _taskService;
+    private readonly UserTaskService _taskService;
 
-        public UserTaskController(UserTaskService taskService)
-        {
-            _taskService = taskService;
-        }
+    public UserTaskController(UserTaskService taskService)
+    {
+        _taskService = taskService;
+    }
 
-        public bool AddTaskForUser(int userId, string description, IResponseModel model)
+    public bool AddTaskForUser(int userId, string description, IResponseModel model)
+    {
+        try
         {
-            string message = GetMessageForModel(userId, description);
-            if (message != null)
-            {
-                model.AddAttribute("action_result", message);
-                return false;
-            }
-
-            return true;
+            var task = new UserTask(description);
+            _taskService.AddTaskForUser(userId, task);
         }
-
-        private string GetMessageForModel(int userId, string description)
+        catch (Exception ex)
```

Comment:
Let's improve this catch block.
Try to imagine what will happen if an unexpected error occurs in UserTaskService. Do you want to return ex.Message to the user in this case?

Reply:
I just added new custom exception type. Please check.

Reply:
Or you could have added multiple catch blocks for multiple types of exceptions 😄

Reply:
Just for training purposes please add a catch block per exception type. This is the usual way of achieving what we want here (although your solution works, too)